lib/src/macos_app.dart

```diff
@@ -65,100 +95,196 @@ class MacosApp extends StatefulWidget {
     this.checkerboardRasterCacheImages = false,
     this.checkerboardOffscreenLayers = false,
     this.showSemanticsDebugger = false,
-    this.debugShowWidgetInspector = false,
     this.debugShowCheckedModeBanner = true,
-    this.inspectorSelectButtonBuilder,
     this.shortcuts,
     this.actions,
-  })  : assert(routeInformationParser != null && routerDelegate != null,
-            'The routeInformationParser and routerDelegate cannot be null.'),
-        assert(supportedLocales.isNotEmpty),
+    this.restorationScopeId,
+    this.themeMode,
+    this.style,
+    this.darkStyle,
+  })  : assert(supportedLocales.isNotEmpty),
         navigatorObservers = null,
-        backButtonDispatcher =
-            backButtonDispatcher ?? RootBackButtonDispatcher(),
         navigatorKey = null,
         onGenerateRoute = null,
-        pageRouteBuilder = null,
         home = null,
         onGenerateInitialRoutes = null,
         onUnknownRoute = null,
         routes = null,
         initialRoute = null,
         super(key: key);
 
-  final Map<Type, Action<Intent>>? actions;
-
-  final BackButtonDispatcher? backButtonDispatcher;
-
-  final TransitionBuilder? builder;
-
-  final bool checkerboardOffscreenLayers;
+  /// {@macro flutter.widgets.widgetsApp.navigatorKey}
```

Comment:
Run dartdoc and see how the macros render - I'm not sure if they will show without a dartdoc options yaml file. If the macros do not render, either remove them or configure dartdoc to render them.

Reply:
But nobody uses dartdoc. When reading these documentations, developers usually use vscode (which works) or api.pub.dev (which also works): https://api.flutter.dev/flutter/cupertino/CupertinoApp/navigatorKey.html

Reply:
pub.dev uses dartdoc to generate documentation for packages published, and I would certainly like to see this published to pub someday 😅

Reply:
This makes no sense. See https://pub.dev/packages/dartdoc#macros

Reply:
I ran dartdoc myself, just to see, and it looks like it doesn't matter after all because since we're exporting material, the material docs get build 😅

<img width="2672" alt="Screen Shot 2021-04-11 at 9 03 09 PM" src="https://user-images.githubusercontent.com/4250470/114328039-767f5780-9b09-11eb-97a5-53313b6f8982.png">

Reply:
Nice!

Reply:
I mean its not really desired behavior, but I can look into that another time 😅